src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IResponseCacheStore.cs

```diff
@@ -2,13 +2,14 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Threading.Tasks;
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
     public interface IResponseCacheStore
     {
-        object Get(string key);
-        void Set(string key, object entry, TimeSpan validFor);
-        void Remove(string key);
+        Task PopulateValueAsync(KeyValuePair keyValuePair);
```

Comment:
I'll compare the KVP vs the Task<object> implementation from the first commit and see what, if any, differences there are.